examples/mocking/lib.rs

```diff
@@ -37,27 +37,19 @@ mod proxy {
 
 #[cfg(test)]
 mod tests {
-    use std::{error::Error, fs, path::PathBuf, rc::Rc};
+    use std::error::Error;
 
     use drink::{
         mock_message,
         runtime::MinimalRuntime,
-        session::{contract_transcode::ContractMessageTranscoder, Session, NO_ARGS},
+        session::{Session, NO_ARGS},
         ContractMock, MockingApi,
     };
 
     use crate::CALLEE_SELECTOR;
 
-    fn transcoder() -> Rc<ContractMessageTranscoder> {
-        Rc::new(
-            ContractMessageTranscoder::load(PathBuf::from("./target/ink/mocking.json"))
-                .expect("Failed to create transcoder"),
-        )
-    }
-
-    fn bytes() -> Vec<u8> {
-        fs::read("./target/ink/mocking.wasm").expect("Failed to find or read contract file")
-    }
+    #[drink::contract_bundle_provider]
+    enum BundleProvider {}
```

Comment:
Not a fan of the DX - it's too magical for me. Personally I prefer the "in your face" approaches.

Reply:
you can't get 'in your face' approach baking such info in the type system... I can see 3 options here:
 - drink provides no support for bundles, i.e. you have to use `ContractBundle` API with string literals, and its your responsibility to care for paths
 - drink provides e.g. mapping from contract name / manifest file to a `ContractBundle` as additional testcase argument (so your tests actually get one argument from the sky), and you depend on string identifiers / paths, for which you are responsible
 - drink provides a typed source of bundles (this PR)

Reply:
The thing with this PR is that I type in code that means nothing if you look at it. What about, sth like this:
```rust
#[drink::test(
  flipper = "path/to/flipper",
  counter = "path/to/counter",
  ...
)]
fn test(bundles: HashMap<String, ContractBundle>) {
 // flipper and counter available here
}
```

Which is basically what you propose in 2)?

Reply:
how does these paths relate to dependencies in `Cargo.toml`? it'd be not cool to specify them for every testcase (actually, macro already knows all the paths)

---

> means nothing if you look at it

means more if you know what it does - every information is implicit (like contract names)

Reply:
> how does these paths relate to dependencies in Cargo.toml? it'd be not cool to specify them for every testcase (actually, macro already knows all the paths)

Hmm, good point. It might overwrite deps in `Cargo.toml`. But also it allows for adding ones that are not defined there. 

> means more if you know what it does - every information is implicit (like contract names)

Exactly. And this is what I don't like about it. 

I'm not gonna block the PR over it ofc.